hack/generator/pkg/astmodel/package_import_set_test.go

```diff
@@ -14,6 +14,10 @@ var (
 	simpleTestRef PackageReference = MakeExternalPackageReference("simple")
 	pathTestRef   PackageReference = MakeExternalPackageReference("package/path")
 
+	// Important for these two package references to have the same version so that they conflict
+	emailTestRef   PackageReference = MakeExternalPackageReference("microsoft.email/v20180801")
```

Comment:
As mentioned above you may want to expand this test to 2x `microsoft.email` versions and 2x `microsoft.network` versions and I think they'll conflict.

Reply:
Oh, that's a good point - if the versions match across the two top-level packages then you'll still get a problem. Maybe the fallback should only be applied to one of the conflicting packages? Would mean you'd need to sort them first for repeatability though.

Reply:
I personally think `emailv20100801` is basically _always_ better to read than `v20180801`, so I sorta think we should maybe just do this all the time for `LocalPackageReference` packages at least.

Reply:
@matthchr I was thinking that too

Reply:
Added fallback to versioned imports, and code to force named imports to be used for local package references.